prody2/viewers/viewer_gnm.py

```diff
@@ -152,35 +154,21 @@ def _viewAllModes(self, paramName):
             title = 'Cross Correlation matrix'
 
         plotter = EmPlotter(mainTitle=title)
-        plot = prody.showMatrix(matrix, origin='lower')
+        plot = prody.showAtomicMatrix(matrix, origin='lower', atoms=self.atoms)
         
         return [plotter] 
 
     def _viewParam(self, paramName):
         if paramName == 'displayModes':
-            # The following two lines display modes.sqlite file
-            # modes =  self.protocol.outputModes
-            # return [ObjectView(self._project, modes.strId(), modes.getFileName())]
-            # The following two lines display modes.xmd file
-            if isinstance(self.protocol, SetOfNormalModes):
-                modes = os.path.dirname(self.protocol[1].getModeFile()) + ".xmd"
-            else:
-                modes =  self.protocol._getPath("modes.xmd")
+            modes = os.path.dirname(self.protocol[1].getModeFile()) + ".xmd"
```

Comment:
This one should be the other one. Protocol can’t be indexed. I think we came across that error together